### ISSUE TITLE
Target of SAD loss

### ISSUE THREAD

**Issue by harryhan618** [cardwing/Codes-for-Lane-Detection#150]
According to the paper, target for output of SAD in a stage should be the output of SAD in the next stage. But in  [`train.lua`](https://github.com/cardwing/Codes-for-Lane-Detection/blob/2b81eaa5f16ad0049c3e83858516264fc8ed525a/ENet-Label-Torch/train.lua#L74), it seems the target for SAD is the segmentation label. Maybe I misunderstand the code?

**Comment by cardwing** [comment]
The targets for block 3 and block 2 are the attention maps of block 4 and block 3, respectively. However, the released code is slightly different from the description (the path of block 2 mimicking block 3 is removed) and I also add the segmentation label to supervise block 4 to obtain much better attention maps.

**Comment by khw321** [comment]
@cardwing About the SAD loss function, you use MSECriterion in the code. But the output attention map's value is 0~1, the target  segmentation label's value is 1~5. And I could only find the MSEloss(block3, seg label) and MSEloss(block4, seg label). Where is the block3 mimicking block4? If I have misunderstood the code?

**Comment by cardwing** [comment]
The added mimicking path is block 3 mimics block 4 and block 4 mimics labels. More details can be found in the model definition in ENet-Label-Torch.

**Comment by khw321** [comment]
Sorry, I can't find the code for block3 mimics block4.     
The model set up just loads the pretrained model and change the channel of existence branch.
And the model has 4 outputs which are segmentation, existence, block3 attention map, block4 attention map.
And the target definition is the code below. target[1] is the segmentation label, target[2] is the existence label.
`      local tmp_target = {}
      table.insert(tmp_target, self.target[1])
      table.insert(tmp_target, self.target[2])
      table.insert(tmp_target, self.target[1])
      table.insert(tmp_target, self.target[1])

      self.target = tmp_target`
So I could only find the MSEloss(block3, seg label) and MSEloss(block4, seg label)
Please review it and let me know if anything is wrong.

**Comment by khw321** [comment]
@cardwing  Could you tell me where is the code for block3 mimics block4? Thank you very much

**Comment by cardwing** [comment]
Sorry that the code only contains block 3 and block 4 mimicking the label. Please add it by yourself.

**Comment by khw321** [comment]
@cardwing Thanks for your reply. About mimicking the label, the output of block 3 and block 4 after power-sum-spatialsoftmax is between 0 and 1. The segmentation label in the torch framework is 1, 2, 3, 4, 5. Could the MSELoss work well? Cause the predict and target is mismatch.
